### PR TITLE
HRINT-2676 - FTSQuerySearchBoosting Java

### PR DESCRIPTION
**Pull request by Drorlev**

### CODE REVIEW

**Review by ml054 — APPROVED** [review]
java part looks good

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBoosting/FTSQuerySearchBoosting.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
+package net.ravendb.demo.textSearch.fTSQuerySearchBasics;
+
+//region Usings
+import net.ravendb.client.documents.session.IDocumentSession;
+//endregion
+import net.ravendb.demo.common.DocumentStoreHolder;
+import net.ravendb.demo.common.models.Employee;
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBoosting/FTSQuerySearchBoosting.java:7]: Shouldn't Usings show also the `DocumentStoreHolder` ?

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBoosting/FTSQuerySearchBoosting.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        List<Employee> employeesWithMatchingTerms;
+
+        try (IDocumentSession session = DocumentStoreHolder.store.openSession()) {
+            //region Step_1
+            employeesWithMatchingTerms = session.query(Employee.class)
+            //endregion
+                //region Step_2
+                .search("Notes", "ph.d.").boost(boost1)
+                .search("Notes", "university").boost(boost2)
+                .search("Notes", "college").boost(boost2)
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBoosting/FTSQuerySearchBoosting.java:27]: boost2 => boost3

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBoosting/metadata.json [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                }
+            ]
+        },
+        {
+            "Title": "Define search criteria",
+            "Slug": "step-2",
+            "DescriptionLines": [
+                "* Use `search()` to search for the specified terms on the specified document field.",
+                "",
+                "* Use `.boost(boostValue)` to set the boost value per term.<br>",
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBoosting/metadata.json:69]: lets have it without the `.`  - similar the `search()`

`.boost(boostValue)` => `boost(boostValue)`

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBoosting/metadata.json [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+                    "Type" : "link",
+                    "Url"  : "https://ravendb.net/docs/article-page/latest/java/indexes/boosting"
+                }
+            ]
+        },
+        {
+            "Title": "Execute the query",
+            "Slug": "step-3",
+            "DescriptionLines": [
+                "* Execute the query by calling `ToList()`.<br>",
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBoosting/metadata.json:89]: ToList() => toList()

**Review by Danielle9897 — CHANGES_REQUESTED** [review]

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/demos.json [hunk 5]
```diff
@@ -100,6 +100,7 @@
         },
         "demos": [
             "fTSWithStaticIndexSingleField",
+            "fTSQuerySearchBoosting",
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/demos.json:103]: make the order similar to the C# order
place `fTSQuerySearchBoosting` before the static index demos

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBoosting/FTSQuerySearchBoosting.java (hunk 2 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBoosting/FTSQuerySearchBoosting.java:27]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBoosting/metadata.json (hunk 3 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBoosting/metadata.json:69]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBoosting/metadata.json (hunk 4 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBoosting/metadata.json:89]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/demos.json (hunk 5 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/demos.json:103]: Done

**Review by Danielle9897 — APPROVED** [review]

**Pull request merged**